pkg/cluster/fetch.go

```diff
@@ -65,10 +65,7 @@ func GetEksdReleaseForCluster(ctx context.Context, cluster *v1alpha1.Cluster, bu
 	if err != nil {
 		return nil, fmt.Errorf("failed fetching versions bundle: %v", err)
 	}
-	eksd, err := fetch(ctx, versionsBundle.EksD.Name, constants.EksaSystemNamespace)
-	if err != nil {
-		return nil, fmt.Errorf("failed fetching EKS-D release for cluster: %v", err)
-	}
+	eksd, _ := fetch(ctx, versionsBundle.EksD.Name, constants.EksaSystemNamespace)
```

Comment:
two thoughts:
- we should validate that the error is the expected error before passing on it; if it's not the expected does not exist error, we shoudl make sure to throw it
- add a debug log statement when it does pass on the error so we know what's happening and why in the future

Reply:
Agreed